kotlin/setup-location-driven-geotriggers/src/main/java/com/esri/arcgisruntime/sample/setuplocationdrivengeotriggers/MainActivity.kt

```diff
@@ -0,0 +1,380 @@
+                            fenceGeotriggerNotificationInfo.geotriggerMonitor.geotrigger.name
+                        )
+                    }
+                }
+            }
+            // if the section has been visited before, get it from the hash map
+        } else {
+            // this garden section has already been visited, show the information again
+            populateUI(
+                sectionsVisited[fenceFeatureName]!!,
+                fenceGeotriggerNotificationInfo.geotriggerMonitor.geotrigger.name
+            )
+        }
+    }
+
+    /**
+     * Removes the [fenceFeatureName] from the garden section or
+     * the points of interest list using the corresponding [geotriggerType]
+     */
+    private fun removeFeatureInformation(fenceFeatureName: String, geotriggerType: String) {
+        if (geotriggerType == sectionGeotriggerName) {
+            sectionButton.text = "N/A"
+        } else {
+            poiList.remove(sectionsVisited[fenceFeatureName])
+            poiListAdapter.notifyDataSetChanged()
+            if (poiList.size == 0) {
+                activityMainBinding.listAvailable.visibility = View.VISIBLE
+            }
+        }
+    }
+
+
+    /**
+     * Populates UI by setting [gardenSection] item based on the [geotriggerType]
+     */
+    private fun populateUI(
+        gardenSection: GardenSection,
+        geotriggerType: String
+    ) {
+        if (geotriggerType == sectionGeotriggerName) {
+            sectionButton.text = gardenSection.title
+            sectionButton.setOnClickListener {
+                if (gardenSection.title == sectionButton.text) {
+                    GardenDescriptionFragment(gardenSection).show(
+                        supportFragmentManager,
+                        "GardenDescriptionFragment"
+                    )
+                }
+            }
+        } else {
+            poiList.add(gardenSection)
+            poiListAdapter.notifyDataSetChanged()
+            activityMainBinding.listAvailable.visibility = View.GONE
+        }
+    }
+
+    /**
+     * Caches the InputStream to a BitmapImage in internal storage.
+     * Returns a String of the image file's absolutePath
+     */
+    private fun saveToInternalStorage(name: String, imageInputStream: InputStream): String {
+
+        val imagePath = File(cacheDir, name.filter { !it.isWhitespace() })
+        var fileOutputStream: FileOutputStream? = null
+        try {
+            fileOutputStream = FileOutputStream(imagePath)
+            // Use the compress method on the BitMap object to write image to the OutputStream
+            val bitmapImage = BitmapFactory.decodeStream(imageInputStream)
+            bitmapImage.compress(Bitmap.CompressFormat.PNG, 100, fileOutputStream)
+        } catch (e: Exception) {
+            e.printStackTrace()
+        } finally {
+            try {
+                fileOutputStream?.close()
+            } catch (e: IOException) {
+                e.printStackTrace()
```

Comment:
I know this is my old code, but let's handle the stackTrace's by writing the message the Log.e